### DISCUSSION TITLE
Episode 213: NumPy & Summed-Area Tables

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#112] [Episode]
Episode Link: https://adspthepodcast.com/2024/12/20/Episode-213.html

Feel free to write comments or ask questions 🙂

**Comment by VincentZalzal** [comment]
This was a really nice and algorithmic episode! I don't know if you discussed this beforehand or if it happened naturally, but I feel that you had better chemistry on this episode, e.g. Bryce seemed genuinely interested in the BQN solution and he was making connections with his own thinking.

Also, related to the summed-area tables, this is what we call integral images in computer vision, because you are integrating (summing) the pixel values over a rectangular area. Among other things, once you have that image, you can compute the sum of the pixel values of any axis-aligned rectangle (not necessarily starting at the origin) by looking up only the four values at the corners of the desired rectangle, making it constant time.

**Reply by codereport** [reply]
Glad to hear! and that is neat to now. A past guest also reached out and said that this algorithm is also used for calculating fifo allocation for trades, so it seems like it pops up all over the place.
